### ISSUE TITLE
Fix multi-select fields in observations passed to iD Editor

### ISSUE THREAD

**Issue by gmaclennan** [digidem/mapeo-desktop#334]
We support multi-select fields in Mapeo which will store the the value as an array, e.g.

```js
{
  type: 'observation',
 //...
  tags: {
    myMultiSelect: ['apples', 'oranges', 'pears']
  }
}
```

However iD only supports tags values as strings. iD does support a multi-select field type called `semiCombo` which stores the result of a multi-select as a semi-colon-separated string.

We could convert arrays to semi-colon-separated strings before passing them to iD, and convert them back if they are edited in iD, but all the ways I can think to do it are fragile:

1. `if (tags.myMultiSelect.includes(';')) newTags.myMultiSelect = tags.myMultiSelect.split(';')` (would get false positive if anyone wrote a `;` in a text field)
2. Lookup field on previous version of observation and if it's an array, `split(';')` (would fail to match multi-select fields which were empty and then edited in iD)
3. Match the observation to the presets on the server and use the field definition to choose which is a multi-select (what happens when the user switches presets? Also the server does not currently have the presets context).
4. Patch something in iD editor to handle this on the front end.
5. Strip any tags that have arrays before passing to iD, and just ignore any value for this tag passed back and replace it with the previous version. Don't show multi-select fields at all in iD (problem is we need multi-select in iD for non-observations i.e. nodes and ways - and it's ok for them to be `;` separated. Solution could be adding a field to the presets which define whether they apply to observations, nodes, lines and/or areas)

The same problem will apply to other non-string field types that we might introduce in the future, e.g. number fields and booleans. MapFilter currently supports these. We don't have a way to collect data currently that would result in a number or boolean value.

**Comment by okdistribute** [comment]
Can't wait until we don't have to deal with iD editor anymore

**Comment by okdistribute** [comment]
Fixed in 5.3.x
